x-pack/qa/oidc-op-tests/src/test/java/org/elasticsearch/xpack/security/authc/oidc/C2IdOpTestCase.java

```diff
@@ -0,0 +1,246 @@
+
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
+
+import org.apache.http.HttpHost;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.protocol.BasicHttpContext;
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.util.EntityUtils;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.core.CheckedFunction;
+import org.elasticsearch.core.PathUtils;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.elasticsearch.xpack.core.common.socket.SocketAccess;
+import org.hamcrest.Matchers;
+import org.junit.BeforeClass;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public abstract class C2IdOpTestCase extends ESRestTestCase {
+    protected static final String TEST_SUBJECT_ID = "alice";
+    private static final String C2OP_PORT = getEphemeralTcpPortFromProperty("oidc-provider", "8080");
+    private static final String LOGIN_API = "http://127.0.0.1:" + C2OP_PORT + "/c2id-login/api/";
+
+    private static final String ES_PORT = getEphemeralTcpPortFromProperty("elasticsearch-node", "9200");
+    private static final String REGISTRATION_URL = "http://127.0.0.1:"
+        + getEphemeralTcpPortFromProperty("oidc-provider", "8080")
+        + "/c2id/clients";
+    // SHA256 of this is defined in x-pack/test/idp-fixture/oidc/override.properties
+    private static final String OP_API_BEARER_TOKEN = "811fa888f3e0fdc9e01d4201bfeee46a";
+
+    private static Path HTTP_TRUSTED_CERT;
+
+    @BeforeClass
+    public static void readTrustedCert() throws Exception {
+        final URL resource = OpenIdConnectAuthIT.class.getResource("/testnode_ec.crt");
+        if (resource == null) {
+            throw new FileNotFoundException("Cannot find classpath resource /testnode_ec.crt");
+        }
+        HTTP_TRUSTED_CERT = PathUtils.get(resource.toURI());
```

Comment:
Interesting. I had the same PathUtils.get call in my PR and it failed in :x-pack:plugin:security:forbiddenApisMain. I assumed your call here would fail too, but it passed CI tests.

It turned out PathUtils#get is restricted in build-tools-internal/src/main/resources/forbidden/es-server-signatures.txt. Those method signatures only apply to main code, not test code. I moved my PathUtils.get call from main code to test code and then it passed.

LGTM